src/Uri.php

```diff
@@ -633,15 +639,37 @@ private static function isNonStandardPort($scheme, $host, $port)
             return false;
         }
 
-        if ($scheme === 'https' && $port !== 443) {
-            return true;
+        $schemes = static::getSchemes();
+
+        return !isset($schemes[$scheme]) || $port !== $schemes[$scheme];
+    }
+
+    /**
+     * Filters the scheme to ensure it is a valid scheme.
+     *
+     * @param string $scheme Scheme name.
+     *
+     * @return string Filtered scheme.
+     */
+    private function filterScheme($scheme)
+    {
+        $scheme = strtolower($scheme);
+        if (strpos($scheme, '://')) {
+            $scheme = str_replace('://', '', $scheme);
         }
 
-        if ($scheme === 'http' && $port !== 80) {
-            return true;
+        if (empty($scheme)) {
+            return '';
+        }
+
+        if (!array_key_exists($scheme, $this->getSchemes())) {
```

Comment:
This is attempting to resolve a static method as an instance method; change to `static::getSchemes()` or `self::getSchemes()`.